### PR TITLE
Added Shows and Top Rated UI and logic

### PR DESCRIPTION
**Pull request by ssvetina**
https://user-images.githubusercontent.com/32631956/179722067-bd421e17-093a-47d6-ad42-64cb588f0d33.mov

In this PR: 
* Added Shows and Top Rated UI and logic

Next steps: 
* Add show details screen

### CODE REVIEW

File: TVShows/Modules/Login/LoginInteractor.swift [hunk 1]
```diff
@@ -43,12 +43,15 @@ extension LoginInteractor: LoginInteractorInterface {
...
+            .do(onSuccess: { [unowned self] data in
+                guard let response = data.response else { return }
+                
+                let userAuth = UserAuth(from: response)
                 if rememberMe {
-                    self.storage.storeUser(me)
+                    self.storage.storeUser(userAuth)
                 }
-                sessionManager.authorizationAdapter = UserAdapter(with: me)
+                sessionManager.authorizationAdapter = UserAdapter(with: userAuth)
```
Comment (z-turk3) [inline TVShows/Modules/Login/LoginInteractor.swift:54]: This is OK now, but you will need to parse the User data also here when you will need to show the users profile.
By user data I mean the name, profile image etc.

**Review by z-turk3 — COMMENTED** [review]
Looks good, left some comments to check. One issue is that the pagination is not working correctly, check my comment below.

File: TVShows/Modules/Login/LoginInteractor.swift (hunk 1 above)
Comment (ssvetina) [reply TVShows/Modules/Login/LoginInteractor.swift:54]: I've meant to cover this in the Settings PR :)

File: TVShows/Modules/Shows/ShowsInteractor.swift [hunk 2]
```diff
@@ -10,11 +10,69 @@
...
+            showNextPage.mapTo(Paging.Event<[Show]>.nextPage),
+            reload.startWith(()).mapTo(Paging.Event<[Show]>.reload)
+        )
+        
+        return Paging.page(
+            make: nextPage,
+            startingWith: [],
+            joining: accumulator,
+            while: hasNext,
+            on: actions.asObservable())
```
[the remark this answers is not in the document]
Comment (ssvetina) [reply TVShows/Modules/Shows/ShowsInteractor.swift:44]: damn, meant to declare them as static... must have been in a hurry and forgot to add :(

**Review by z-turk3 — APPROVED** [review]
Good job 👍

**Pull request merged**